twine/package.py

```diff
@@ -44,6 +44,14 @@
 MetadataValue = Union[str, Sequence[str]]
 
 
+def _safe_name(name: str) -> str:
+    """Convert an arbitrary string to a standard distribution name.
+
+    Any runs of non-alphanumeric/. characters are replaced with a single '-'.
+    """
```

Comment:
I think it's helpful to give context for code ported from other projects:

```suggestion
    Any runs of non-alphanumeric/. characters are replaced with a single '-'.
    
    Copied from pkg_resources.safe_name for compatibility with warehouse.
    See https://github.com/pypa/twine/issues/743.
    """
```

Reply:
Good idea! Thanks for the suggestion 👍